### ISSUE TITLE
Add exception handling for calls to WC_Data_Store

### ISSUE THREAD

**Issue by reykjalin** [Automattic/woocommerce-payments#1131]
`WC_Data_Store` can throw exceptions that we are not catching, e.g. here: https://github.com/Automattic/woocommerce-payments/blob/bc7e5527c38582ba7f727fb7341874fdc962d469/includes/class-wc-payments.php#L128

We faced a fatal error in the Stripe plugin in a similar situation, but @dechov noted in slack (p1606199393338000-slack-team-fractal?thread_ts=1605669204.259700&cid=C01BZUL57SQ) that the same situation cannot occur here because WCPay isn't loaded when WC-Admin is disabled via the `woocommerce_admin_disabled` hook.

We should still take the time to handle these exceptions, since

1. That's just good practice 🙂 ; and
1. `WC_Data_Store` might throw exceptions in situations we don't know about yet.

cc @v18

**Comment by vbelolapotkov** [comment]
@dwainm sending this one to your team's maintenance backlog. I've assigned the issue to you for this, please take care of adding it to the team board.

**Comment by dwainm** [comment]
Added it to our refinement flow.

**Comment by jessepearson** [comment]
_tl:dr; if you use the `woocommerce_admin_disabled` filter and set it to `true`, WooPayments doesn't load._

Going to close this due to if the filter `woocommerce_admin_disabled` is set to `true`, then the code exits before getting to the problematic point.

This is done in the `WC_Payments` class:

https://github.com/Automattic/woocommerce-payments/blob/36ae2f07c8ddc09048203d1e27a03a7bee9be14b/includes/class-wc-payments.php#L346-L348

The `has_valid_dependencies` method in the [`WC_Payments_Dependency_Service` class](https://github.com/Automattic/woocommerce-payments/blob/36ae2f07c8ddc09048203d1e27a03a7bee9be14b/includes/class-wc-payments-dependency-service.php#L18) calls `get_invalid_dependencies` in the same class. If the returned array is not empty, then `false` is returned. The `get_invalid_dependencies` method calls `is_wc_admin_enabled` in the same class checks the `woocommerce_admin_disabled` filter and if it returns `true`, then an invalid dependency is reported to be return to the `has_valid_dependencies` check.
